### PR TITLE
82 mediator coordination stateful keylist query

### PR DESCRIPTION
**Pull request by ndefokou**
Hi @IngridPuppet @Christiantyemele,

Please review this PR

### CODE REVIEW

File: mediator-coordination/src/web/coord/handler/stateful.rs
Comment (IngridPuppet) [inline mediator-coordination/src/web/coord/handler/stateful.rs:1]: Please note that the mediation-coordination crate was moved under `crates/plugins/mediation-coordination`. That said, your updates should be moved there too.
image

**Review by IngridPuppet — CHANGES_REQUESTED** [review]

File: crates/plugins/mediator-coordination/src/web/coord/handler/stateful.rs [hunk 1]
```diff
@@ -366,6 +433,58 @@ mod tests {
...
+        )
+        .header("return_route".into(), json!("all"))
+        .to(global::_mediator_did(&state))
+        .from(global::_edge_did())
+        .finalize();
+
+        // Process request
+        let err = process_plain_keylist_update_message(Arc::clone(&state), message)
+            .await
+            .unwrap_err();
```
Comment (IngridPuppet) [inline crates/plugins/mediator-coordination/src/web/coord/handler/stateful.rs:478]: This negative test is testing something else than `process_plain_keylist_query_message`.

File: .gitignore [hunk 2]
```diff
@@ -22,6 +22,12 @@ Cargo.lock
 # Mac OS X Finder metadata
 .DS_Store
 
+# Ignore .env files in the root directory
+*.env
+
+# Ignore .env files in all subdirectories
+**/*.env
+
```
Comment (IngridPuppet) [inline .gitignore:30]: Could you remove this? We'll re-discuss it at a later point.

**Review by IngridPuppet — CHANGES_REQUESTED** [review]

File: crates/plugins/mediator-coordination/src/web/coord/handler/stateful.rs (hunk 1 above)
Comment (ndefokou) [reply crates/plugins/mediator-coordination/src/web/coord/handler/stateful.rs:478]: Should i remove it?

Comment (IngridPuppet) [reply crates/plugins/mediator-coordination/src/web/coord/handler/stateful.rs:478]: No, replace it with a proper negative test.

File: crates/plugins/mediator-coordination/src/web/coord/handler/stateful.rs [hunk 3]
```diff
@@ -366,6 +433,56 @@ mod tests {
...
+        .finalize();
+
+        // Process request
+        let response = process_plain_keylist_query_message(Arc::clone(&state), message)
+            .await
+            .unwrap();
+
+        assert_eq!(response.type_, KEYLIST_2_0);
+        assert_eq!(response.from.unwrap(), global::_mediator_did(&state));
+        assert_eq!(response.to.unwrap(), vec![global::_edge_did()]);
```
Comment (Christiantyemele) [inline crates/plugins/mediator-coordination/src/web/coord/handler/stateful.rs:457]: @ndefokou can you assert the Message body as it is what we are interested in

**Review by Christiantyemele — CHANGES_REQUESTED** [review]

**Review by Christiantyemele — COMMENTED** [review]
@ndefokou can you assert the Message body as it is what we are interested in

**Review by IngridPuppet — APPROVED** [review]

**Comment by IngridPuppet** [comment]
> @ndefokou can you assert the Message body as it is what we are interested in

Hi @Christiantyemele - Doesn't the definition of `_assert_delegate_handler_err` answer this point?

**Review by Christiantyemele — APPROVED** [review]

**Review by Christiantyemele — APPROVED** [review]

**Pull request merged**